Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp

```diff
@@ -137,6 +138,21 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               M_xface(i, j, k, 2) += dt * (-PhysConst::mu0 * mag_gamma_interp) * ( M_xface(i, j, k, 0) * Hy_eff - M_xface(i, j, k, 1) * Hx_eff)
                 + dt * Gil_damp * ( M_xface(i, j, k, 0) * ( M_xface(i, j, k, 2) * Hx_eff - M_xface(i, j, k, 0) * Hz_eff)
                 - M_xface(i, j, k, 1) * ( M_xface(i, j, k, 1) * Hz_eff - M_xface(i, j, k, 2) * Hy_eff));
+
+          // temporary normalized magnitude of M_xface field at the fixed point
+              Real M_normalized = sqrt( pow(M_xface(i, j, k, 0),2.0) + pow(M_xface(i, j, k, 1),2.0) + pow(M_xface(i, j, k, 2),2.0) ) /
```

Comment:
Could you comment here that we need to re-investigate the way we do Ms interp, in case in the future we encounter the case where Ms changes across two adjacent cells that you are doing interp?

Reply:
Ok!